bindsnet/pipeline/__init__.py

```diff
@@ -225,14 +228,24 @@ def plot_reward(self) -> None:
         """
         Plot the accumulated reward for each episode.
         """
+        # Compute moving average
+        if self.reward_window is not None:
+            # Ensure window size > 0 and < size of reward list
+            window = max(min(len(self.reward_list), self.reward_window), 0)
+
+            # Fastest implementation of moving average
+            reward_list_ = pd.Series(self.reward_list).rolling(window=window, min_periods=1).mean().values
+        else:
+            reward_list_ = self.reward_list[:]
+
         if self.reward_im is None and self.reward_ax is None:
             self.reward_im, self.reward_ax = plt.subplots()
             self.reward_ax.set_title('Accumulated reward')
             self.reward_ax.set_xlabel('Episode')
             self.reward_ax.set_ylabel('Reward')
-            self.reward_plot, = self.reward_ax.plot(self.reward_list)
+            self.reward_plot, = self.reward_ax.plot(reward_list_)
```

Comment:
If `self.reward_window` is `None`, is `reward_list_` created?

Reply:
Yes, that is done in the above if-else statement. If `self.reward_window` is `None`, `reward_list_ = self.reward_list[:]`.

Reply:
Nice, I missed that for some reason.